### PR TITLE
refactor(admin): convert rest of components to TS

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* converts Providers
* converts AuthenticatedApp
* fixes redux types
* converts RBACProvider (yes there's also one in the admin package!)

### Why is it needed?

* TS

### Related issue(s)/PR(s)

* resolves CONTENT-1920

### CODE REVIEW

File: packages/core/admin/admin/src/components/AuthenticatedApp.tsx [hunk 1]
```diff
@@ -1,58 +1,94 @@
...
     { data: appInfos, status },
     { data: tagName, isLoading },
     { data: permissions, status: fetchPermissionsStatus, refetch, isFetching },
     { data: userRoles },
   ] = useQueries([
-    { queryKey: 'app-infos', queryFn: fetchAppInfo },
+    {
+      queryKey: 'app-infos',
+      async queryFn() {
+        const { data } = await get('/admin/information');
```
Comment (gu-stav) [inline packages/core/admin/admin/src/components/AuthenticatedApp.tsx:46]: Do we not want to infer the response shape here?

File: packages/core/admin/admin/src/components/AuthenticatedApp.tsx [hunk 2]
```diff
@@ -1,58 +1,94 @@
...
+
+        return data.data;
+      },
+    },
     {
       queryKey: 'strapi-release',
-      queryFn: () => fetchStrapiLatestRelease(toggleNotification),
+      async queryFn() {
+        try {
+          const res = await fetch('https://api.github.com/repos/strapi/strapi/releases/latest');
```
Comment (gu-stav) [inline packages/core/admin/admin/src/components/AuthenticatedApp.tsx:55]: Same.

File: packages/core/admin/admin/src/components/AuthenticatedApp.tsx (hunk 1 above)
Comment (joshuaellis) [reply packages/core/admin/admin/src/components/AuthenticatedApp.tsx:46]: yes! thanks for the reminder, I thought we had `noImplicitAny` on to stop this 🤔

File: packages/core/admin/admin/src/components/AuthenticatedApp.tsx [hunk 3]
```diff
@@ -43,7 +46,21 @@ const AuthenticatedApp = () => {
...
+            | 'communityEdition'
+            | 'dependencies'
+            | 'useYarn'
+            | 'projectId'
+            | 'strapiVersion'
+            | 'nodeVersion'
+          >;
+        }
+
+        const { data } = await get<InformationResponse>('/admin/information');
```
Comment (gu-stav) [inline packages/core/admin/admin/src/components/AuthenticatedApp.tsx:63]: Should this rather use `ApiResponse<InformationResponse>`?

**Review by gu-stav — DISMISSED** [review]

**Comment by joshuaellis** [comment]
Sorry i had to fix tests :(

**Review by gu-stav — APPROVED** [review]

**Pull request merged**